Move the registration of CPython builtin modules to BuiltinRegistry

PaliC opened the pull request:
Summary: leverages BuiltinRegistry to register the CPython standard C modules. The standard C modules moved are in the FOR_EACH macro

Test Plan:
buck test mode/opt //caffe2/torch/csrc/deploy/interpreter:test_builtin_registry

buck test mode/opt //caffe2/torch/csrc/deploy:test_deploy

Differential Revision: D31848547

shunting314 reviewed (changes requested):
Overall looks pretty good to me!
Just one thing about code structure. Can we put all the registration code for frozenpython to its separate file since it gets bigger? Check an example of how we do that for numpy [here](https://github.com/pytorch/pytorch/blob/master/torch/csrc/deploy/interpreter/register_numpy.cpp)

PaliC committed:
Move the registration of CPython builtin modules to BuiltinRegistry (#67085)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/67085

leverages BuiltinRegistry to register the CPython standard C modules. The standard C modules moved are in the FOR_EACH macro

Test Plan:
buck test mode/opt //caffe2/torch/csrc/deploy/interpreter:test_builtin_registry

buck test mode/opt //caffe2/torch/csrc/deploy:test_deploy

Differential Revision: D31848547

fbshipit-source-id: 9e33d0c0a350cf62467ba3b1efea3dda685d392e
